mltensor-explainer.md

```diff
@@ -0,0 +1,426 @@
+  mlContext.readTensor(outputMlTensor3, outputArrayBuffer3);
+]);
+```
+
+### Chained Inference
+
+You may notice another benefit of the code snippet above: each call to `dispatch()` does not require an `await`. 
+
+```js
+// Current approach to execute models repeatedly, requiring many round-trips to script
+
+// The input and output buffers are transferred. We must wait for the buffers to be returned -
+// after the graph is executed and output buffers are copied out - before executing the graph
+// with these inputs again.
+await mlContext.compute(graph1, {'input': imageAsArrayBuffer}, {'output': outputArrayBuffer1});
+// The ML context sits idle while control returns to script...
+await mlContext.compute(graph2, {'input': imageAsArrayBuffer}, {'output': outputArrayBuffer2});
+// ...which just wants to invoke the ML context again. There has to be a better way!
+await mlContext.compute(graph3, {'input': imageAsArrayBuffer}, {'output': outputArrayBuffer3});
+```
+
+Using `MLTensor` enables a programming model similar to [WebGPU's](https://www.w3.org/TR/webgpu/#programming-model). Tasks are posted to the ML context's [timeline](#timelines) and are executed as the ML context sees fit - so far as data dependencies are respected such that each `MLTensor` is guaranteed to be modified in the order the methods using the tensor are called from script. In this example, the ML context should be working continuously from the `writeTensor()` call until the work for the last `readTensor()` completes. Better utilization of the ML context will result in significantly better throughput.
+
+```js
+// Proposed approach to queue tasks to the ML context timeline
+
+// Post a task to the ML context timeline to allocate and zero out a tensor,
+// then return to script.
+const imageAsMlTensor = await mlContext.createTensor({..., usage: MLTensorUsage.WRITE});
+
+// Post a task to the ML context timeline to write to the tensor. Note that we do
+// not await completion of this write. The ML context will ensure any operations
+// which depend on the contents of `imageAsMlTensor` will queue behind this task.
+mlContext.writeTensor(imageAsMlTensor, imageAsArrayBuffer);
+
+// Post a task to the ML context timeline to execute the graph. The ML context will
+// ensure this queues behind the write above.
+mlContext.dispatch(graph1, {'input': imageAsMlTensor}, {'output': outputMlTensor1});
+// Post another task. Since input tensors will not be modified by graph execution,
+// the ML context may choose to execute in parallel to the dispatch above.
+mlContext.dispatch(graph2, {'input': imageAsMlTensor}, {'output': outputMlTensor2});
+// Post another task, which may also execute in parallel.
+mlContext.dispatch(graph3, {'input': imageAsMlTensor}, {'output': outputMlTensor3});
+
+// Post tasks to read the output tensors. These tasks will queue behind the
+// respective dispatch() calls using each tensor.
+const outputs = await Promise.all([
+    outputMlTensor1,
```

Comment:
I'm kind of confused by this example since `outputMlTensor1` is not declared, but is passed in above. Is it a promise?

Reply:
`outputMlTensor1` is an `MLTensor`. Updated this code snippet to make it more obvious what's happening here